lib/improver/wind_gust_diagnostic.py

```diff
@@ -0,0 +1,227 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing plugin for WindGustDiagnositic."""
+
+import warnings
+
+import iris
+from iris import FUTURE
+
+from improver.utilities.cube_manipulation import merge_cubes
+
+FUTURE.netcdf_promote = True
+
+
+class WindGustDiagnostic(object):
+
+    """Plugin for calculating wind-gust diagnostic.
+
+    In the model a shear-driven turbulence parameterization is used to
+    estimate wind gusts but in convective situations this can over-estimate the
+    convective gust.
+    This diagnostic takes the Maximum of the values at eachgrid point of
+        a chosen percentile of the wind-gust forecast and
+        a chosen percentile of the wind-speed forecast
+    to produce a better estimate of wind-gust.
+    For example a typical wind-gust could be MAX(gust(50%),windspeed(95%))
+    an extreme wind-gust forecast could be MAX(gust(95%), windspeed(100%))
+
+    See
+    https://github.com/metoppv/improver/files/1244828/WindGustChallenge_v2.pdf
+    for a discussion of the problem and proposed solutions.
+    """
+
+    def __init__(self, percentile_gust, percentile_windspeed):
+        """
+        Create a WindGustDiagnostic plugin for a given set of percentiles.
+
+        Args:
+            percentile_gust: float
+                Percentile value required from wind-gust cube.
+            percentile_windspeed: float
+                Percentile value required from wind-speed cube.
+
+        """
+        self.percentile_gust = percentile_gust
+        self.percentile_windspeed = percentile_windspeed
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<WindGustDiagnostic: wind-gust perc='
+                '{0:3.1f}, wind-speed perc={1:3.1f}>'
+                .format(self.percentile_gust, self.percentile_windspeed))
+        return desc
+
+    def add_metadata(self, cube):
+        """Add metadata to cube for windgust diagnostic.
+
+        Args:
+            cube: iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result : iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.standard_name = "wind_speed_of_gust"
+        result.long_name = "wind_gust_diagnostic"
```

Comment:
Metadata is not defined for this function in the ticket so I have made some up so that the cube is easily identified. If we decide that the Metadata is inappropriate then we can change it but until then I see no problem with this.